README.md

```diff
@@ -173,6 +173,34 @@ Using the `Parameters` classes is not required, but they can help in a number of
 - Provides a more natural way to construct parameters
 - Facilitates ASP.NET Core model binding (bind an HTML form to a `Parameters` instance)
 
+## Debugging
+
+### View HTTP Requests
+
+Users can audit the HTTP requests and responses being made by the library by setting the `Hooks` property of a `ClientConfiguration` with a set of event handlers. Available handlers include:
+
+- `OnRequestBeforeExecution` - Called before an HTTP request is made. An `RequestBeforeExecutionEventArgs` object is passed to the handler, which contains the `HttpRequestMessage` that will be sent to the server.
+  - The `HttpRequestMessage` at this point is configured with all expected data (headers, body, etc.). Modifying the `HttpRequestMessage` will NOT affect the actual request that is sent to the server.
+- `OnRequestResponseReceived` - Called after an HTTP request is made. An `RequestResponseReceivedEventArgs` object is passed to the handler, which contains the `HttpResponseMessage` that was received from the server.
+
+Users can interact with these details in their callbacks as they see fit (e.g. logging).
+
+```csharp
+Client client = new Client(new ClientConfiguration("EASYPOST_API_KEY")
+{
+    Hooks = new Hooks {
+        OnRequestBeforeExecution = (_, args) => {
+            // Interact with the HttpRequestMessage here via args.Request
+            System.Console.WriteLine($"Making HTTP call to {args.Request.RequestUri}");
+        },
+        OnPostRequest = (_, args) => {
+            // Interact with the HttpResponseMessage here via args.Response
+            System.Console.WriteLine($"Received HTTP response with status code {args.Response.StatusCode}");
+        },
```

Comment:
I'd suggest in the example to define these separately and then pass them into the hooks as I can see most people not doing this inline as the logic could be long. Would make this a bit cleaner